os/pm/pm_changestate.c

```diff
@@ -227,23 +118,10 @@ int pm_changestate(enum pm_state_e newstate)
 	 * drivers may refuse the state change.
 	 */
 	if ((newstate != PM_RESTORE) && (newstate != g_pmglobals.state)) {
-		ret = pm_prepall(newstate);
-		if (ret != OK) {
-			/* One or more drivers is not ready for this state change.  Revert to
-			* the preceding state.
-			*/
-
-			g_pmglobals.recommended = g_pmglobals.state;
-			g_pmglobals.btime = clock_systimer();
-			goto EXIT;
-		}
-		/* All drivers have agreed to the state change (or, one or more have
-		* disagreed and the state has been reverted).  Set the new state.
-		*/
-		pm_changeall(newstate);
 		g_pmglobals.state = newstate;
+		g_pmglobals.stime = clock_systimer();
 	}
-EXIT:
+
 	/* Restore the interrupt state */
 	leave_critical_section(flags);
 	return ret;
```

Comment:
You can directly return OK here.  Also please remove pm_checkstate conditional check from pm_idle & other bsp code